src/libsyntax_ext/source_util.rs

```diff
@@ -83,7 +83,13 @@ pub fn expand_include<'cx>(cx: &'cx mut ExtCtxt<'_>, sp: Span, tts: TokenStream)
     }
     impl<'a> base::MacResult for ExpandResult<'a> {
         fn make_expr(mut self: Box<ExpandResult<'a>>) -> Option<P<ast::Expr>> {
-            Some(panictry!(self.p.parse_expr()))
+            let r = Some(panictry!(self.p.parse_expr()));
+            if self.p.token != token::Eof {
+                self.p.sess.span_diagnostic.span_fatal(
```

Comment:
Why fatal?

Reply:
As I mentioned in the PR description, I think it's an open question what we want here, a fatal diagnostic was just the easiest thing to immediately implement.